tests/Feature/MutualTlsTest.php

```
@@ -0,0 +1,310 @@
+    $this->setProperty($client, 'token_endpoint_auth_methods_supported', [ClientAuthMethod::TLS_CLIENT_AUTH]);
+
+    expect($this->invokeMethod($client, 'getClientAuthMethod'))->toBeNull()
+        ->and($client->usesMutualTlsClientAuth())->toBeFalse();
+});
+
+test('no client secret is sent under mutual tls', function () {
+    $data = ['grant_type' => 'client_credentials', 'client_secret' => 'should-be-removed'];
+
+    expect($this->invokeMethod(mtlsClient(), 'applyMutualTlsClientAuth', [&$data]))->toBeTrue()
+        ->and($data)->not->toHaveKey('client_secret')
+        ->and($data['client_id'])->toBe('mtls-client');
```

PHPStan (2.2.7)

[error] Offset 'client_id' does not exist on array{grant_type: 'client_credentials', client_secret: 'should-be-removed'}.


(offsetAccess.notFound)